src/data/fun_like/basic.lean

```diff
@@ -57,12 +57,16 @@ the axioms of your new type of morphisms.
 Continuing the example above:
 
 ```
+set_option old_structure_cmd true
+
 /-- `my_hom_class F A B` states that `F` is a type of `my_class.op`-preserving morphisms.
 You should extend this class when you extend `my_hom`. -/
 class my_hom_class (F : Type*) (A B : out_param $ Type*) [my_class A] [my_class B]
   extends fun_like F A (λ _, B) :=
 (map_op : ∀ (f : F) (x y : A), f (my_class.op x y) = my_class.op (f x) (f y))
 
+set_option old_structure_cmd false
+
```

Comment:
I'd argue it's nicer to put `set_option`s with limited scope in a section, that means we don't accidentally disable it for other declarations.
```suggestion
section
set_option old_structure_cmd true

/-- `my_hom_class F A B` states that `F` is a type of `my_class.op`-preserving morphisms.
You should extend this class when you extend `my_hom`. -/
class my_hom_class (F : Type*) (A B : out_param $ Type*) [my_class A] [my_class B]
  extends fun_like F A (λ _, B) :=
(map_op : ∀ (f : F) (x y : A), f (my_class.op x y) = my_class.op (f x) (f y))

end
```